js/app/interfaces/scrolling.js

```diff
@@ -0,0 +1,46 @@
+// Copyright 2018 Endless Mobile, Inc.
+
+const Gtk = imports.gi.Gtk;
+const Lang = imports.lang;
+
+const HistoryStore = imports.app.historyStore;
+const Module = imports.app.interfaces.module;
+
+/**
+ * Interface: Scrolling
+ */
+var Scrolling = new Lang.Interface({
+    Name: 'Scrolling',
+    GTypeName: 'EknScrolling',
+    Requires: [Gtk.ScrolledWindow, Module.Module],
+
+    Slots: {
+        /**
+         * Slot: content
+         * Where to put the content
+         */
+        'content': {},
+    },
+
+    _interface_init: function () {
+        HistoryStore.get_default().connect('changed', this._return_to_top.bind(this));
+    },
+
+    // return scroll position to the top of the window
+    _return_to_top: function () {
+        let store = HistoryStore.get_default();
+        let current = store.get_current_item();
+        let previous = store.get_direction() === HistoryStore.Direction.FORWARDS ?
+            store.get_previous_item() : store.get_next_item();
+
+        // if switching to or from a lightbox, don't reset
+        if ((current && current.media_model) || (previous && previous.media_model)) {
```

Comment:
I think you want XOR here. This code will reset the scroll if you use the arrows to page to another image while in the lightbox.

Unfortunately there is no logical XOR operator in JS but you can fake it with: `if (!(current && current.media_model) !== !(previous && previous.media_model))`

Reply:
> This code will reset the scroll if you use the arrows to page to another image while in the lightbox.

Switching between images in a lightbox does not interact with the historyStore. It won't modify the historyStore, so it can't be navigated with the page buttons (and can't reset the scroll).

However, the change should be fine, and it would make it more robust for possible changes in the future.